### PR TITLE
Fix analyze ignorehttpserrors flag

### PR DESCRIPTION
**Pull request by floo51**
- Upgrade playwright to latest
- Lock playwright version to avoid future incompatibilities
- Fix tests mechanism + some tests
- Fix ignoreHTTPSErrors on analyze command
- Test it

### CODE REVIEW

File: .greenframe.e2e.multiple.yml [hunk 1]
```diff
@@ -9,7 +9,3 @@ baseURL: 'https://greenframe.io'
 samples: 2
 distant: false
 projectName: 'GreenFrame'
-containers:
```
Comment (fzaninotto) [inline .greenframe.e2e.multiple.yml:12]: I believe the "multiple" scenario was designed to test multiple containers

File: e2e/index.js [hunk 2]
```diff
@@ -2,34 +2,36 @@
...
                 `${BASE_COMMAND} -C ./.greenframe.e2e.yml`
             );
 
             expect(stdout).toContain('✅ main scenario completed');
             expect(stdout).toContain('The estimated footprint is');
-            expect(stdout).toContain('Check the details of your analysis at');
-            expect(error).toBeUndefined();
-        });
 
-        it('E2E: Should run an analysis and fail because account is suspended', async () => {
```
Comment (fzaninotto) [inline e2e/index.js:20]: Why did you remove this one?

File: e2e/index.js [hunk 3]
```diff
@@ -89,48 +92,36 @@ describe('E2E: analyze command locally', () => {
             }
         });
 
-        it('E2E: Should fail because user has reached the project limit', async () => {
```
Comment (fzaninotto) [inline e2e/index.js:92]: do not remove

File: e2e/index.js [hunk 4]
```diff
@@ -77,7 +79,8 @@ describe('E2E: analyze command locally', () => {
             expect(error).toBeUndefined();
         });
 
-        it('E2E: Should fail because "container_broken" is not a running container', async () => {
+        // we need to setup a mock dev environment to enable this test
+        it.skip('E2E: Should fail because "container_broken" is not a running container', async () => {
```
Comment (fzaninotto) [inline e2e/index.js:83]: linter warning for skipped tests

File: e2e/index.js [hunk 5]
```diff
@@ -89,48 +92,36 @@ describe('E2E: analyze command locally', () => {
...
             expect(stdout).toContain('The estimated footprint is');
-            expect(stdout).toContain('Check the details of your analysis at');
             expect(error).toBeUndefined();
         });
     });
 });
 
-describe('E2E: analyze command distant', () => {
+// we need to setup a mock greenframe.io environment to enable this test
+describe.skip('E2E: analyze command distant', () => {
```
Comment (fzaninotto) [inline e2e/index.js:120]: same

File: package.json [hunk 6]
```diff
@@ -108,15 +108,15 @@
...
+        "watch": "rm -rf ./dist && mkdir -p ./dist/bash && cp -R ./src/bash ./dist/ && tsc --watch",
+        "build": "rm -rf ./dist && tsc && mkdir -p ./dist/bash &&cp -R ./src/bash ./dist/",
         "dev": "ts-node-dev src/index.ts",
         "postpack": "rm -f oclif.manifest.json",
         "posttest": "eslint .",
         "prepack": "yarn build && oclif manifest && oclif readme",
         "test-unit": "jest ./src",
         "test-watch": "yarn test-unit --watch",
-        "test-e2e": "wait-on http://localhost:3006 && jest ./e2e/index.js --testTimeout 120000",
+        "test-e2e": "jest ./e2e/index.js --testTimeout 120000",
```
Comment (fzaninotto) [inline package.json:119]: again, this is no longer an e2e test

**Review by fzaninotto — CHANGES_REQUESTED** [review]
I'm not a fan of removing useful tests. I suggest you split the e2e tests between one file that tests without the server, and the other with the server (and that one is probably skipped for now)

File: e2e/index.js [hunk 7]
```diff
@@ -2,34 +2,36 @@
 const util = require('node:util');
 const exec = util.promisify(require('node:child_process').exec);
 
-const BASE_COMMAND = `GREENFRAME_SECRET_TOKEN=API_TOKEN API_URL=http://localhost:3006 ./bin/run analyze`;
+const BASE_COMMAND = `./bin/run analyze`;
```
[the remark this answers is not in the document]
Comment (floo51) [reply e2e/index.js:5]: That's still an e2e tests, it tests the app workflow from beginning to end, when used without the SaaS service.

**Comment by fzaninotto** [comment]
You must also bump the package version to have it republished

**Review by fzaninotto — APPROVED** [review]

**Pull request merged**